From 1f7c848caa0e18aa81350c7c26628e4ec4725f57 Mon Sep 17 00:00:00 2001
Message-Id: <1f7c848caa0e18aa81350c7c26628e4ec4725f57.1430166434.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Mon, 6 Apr 2015 16:18:42 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Fix for sporadic display
 blank-out during Lock/Unlock screen.

Display planes need to enabled before the encoder in case of DSI as per HW recommendation.
Without this change, display blanks out sporadically during
scenarios like continuous lock/unlock screen. Squash this patch to:
	commit d4e59a3857055dceee16463647a226f6bf4f4356
	Author: vkorjani <vikas.korjani@intel.com>
	Date:   Mon Mar 2 15:00:26 2015 +0530
	Change-Id: I27649baa4adf8d7fa2f7917290e47dd42b04f9bb
    FOR_UPSTREAM [VPG] drm/i915: Move port enable to after pipe enable in DSI sequence.

Issue: GMINL-7665
Change-Id: I08da2a90c31431effbbb6e459e4c7395bc45021f
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index db24d6f..904d06a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5395,10 +5395,17 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 
 	intel_enable_pipe(intel_crtc);
 
+	/* For DSI, enable encoder after planes */
 	for_each_encoder_on_crtc(dev, crtc, encoder)
-		encoder->enable(encoder);
+		if (encoder->type != INTEL_OUTPUT_DSI)
+			encoder->enable(encoder);
 
 	intel_crtc_enable_planes(crtc);
+
+	for_each_encoder_on_crtc(dev, crtc, encoder)
+		if (encoder->type == INTEL_OUTPUT_DSI)
+			encoder->enable(encoder);
+
 	intel_update_drrs(dev);
 
 	drm_crtc_vblank_on(crtc);
-- 
1.7.9.5

